Add minifollowups module to pycbc.workflow

cmbiwer opened the pull request:
This adds a skeleton module for minifollowups to ``pycbc.workflow``.

Note that the module will not run unless ``[workflow-minifollowups]`` is in the ini file, if it is not, then it will just exit the module and continue with the workflow generation. This section has not been added to the example ini files because there still is more work to do.

I provide an example of how to add an executable to the module.

https://github.com/ligo-cbc/pycbc/compare/ligo-cbc:master...cmbiwer:mf_pr?expand=1#diff-6d5d67ddd622a3b42cb10c895f203f90R62 - Create a class for creating nodes for the executable

https://github.com/ligo-cbc/pycbc/compare/ligo-cbc:master...cmbiwer:mf_pr?expand=1#diff-6d5d67ddd622a3b42cb10c895f203f90R50 - Then create an instance and create a node for the n-th loudest job.

At the end you should get a page under the Results section that looks like: https://sugar-jobs.phy.syr.edu/~cbiwer/pycbc_test_workflow/mf_test//7._result/7.1_loudest_event_1_of_10/

TJ - It looks like you have a executable that creates a separate file for all the follow ups. If that's the case, then the loop can be deleted and you just need to add that executable in as well.

cmbiwer commented:
Fixed merged conflicts so forced pushed a single clean commit that adds module.

cmbiwer commented:
Travis check failed with ``error: timed out``. Looks like the check just took too long.

duncan-brown commented:
Chris will add mass bins to the patch.

titodalcanton commented:
I restarted the travis build, I think it was one of the usual network glitches.

tjmassin commented:
I'm going through the example to make sure I understand how the settings are all being passed down to the executables. Is the plan to have this module pull the list of follow-up executables from the ini file under the [workflow-minifollowups] section?

spxiwh commented on pycbc/workflow/minifollowups.py:
```diff
@@ -0,0 +1,86 @@
+                 ifo=None, out_dir=None, tags=[], universe=None):
+        super(MinifollowupsTableExecutable, self).__init__(cp, exe_name, universe, ifo, out_dir, tags=tags)
+
+    def create_node(self, segment, coinc_file, tmpltbank_file, data_type, n):
```
Ideally this function would be documented, but that can happen later. Please can you change the variable name "n" to something meaningful though? I have no idea what n means?

spxiwh commented on pycbc/workflow/minifollowups.py:
```diff
@@ -0,0 +1,86 @@
+
+        # add options
+        node.add_opt('--data-type', data_type)
+        node.add_opt('--n', n)
```
Also here, a command line argument "--n" is not clear, please change this.

spxiwh commented:
Hi Chris,

Can you update me on the status of this? There are unanswered requests from Duncan and TJ above. I made a couple of comment in-line about the code. Once all this is addressed this will be good to merge.

cmbiwer commented:
Duncan, I've moved the call to minifollowups so that its done for each mass bin.

> Is the plan to have this module pull the list of follow-up executables from the ini file under the [workflow-minifollowups] section?

TJ, follow-up executables go under the ``[executables]`` section. At the moment ``[workflow-minifollowups]`` will just tell the workflow generator how many of the loudest events to followup in each bin, eg. to followup the 10 loudest in each bin set:
```
[workflow-minifollowups]
num-events = 10
```

Ian, I've add a comment and renamed the option.

cmbiwer commented:
Looks like Travis failed with:
```
File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/PyCBC-442f83-py2.7.egg/pycbc/results/legacy_grb.py", line 28, in <module>
    from pylal import antenna,git_version
```

duncan-brown commented:
Yes, this is the GRB code merge. Working on it now.

ahnitz commented:
There is no minifollowups table executable as far as I can see, was that an omission, or do we need to still write that, and this code should not be turned on?

duncan-brown commented:
I think you need to write that Alex.

tjmassin commented:
I have a script that reads in the coinc trigger files, single IFO trigger files, and bank file and strips away the interesting statistics for the N loudest events. Right now it just writes them to a new HDF file, but I'll talk to Alex about tweaking the output to generate a table of events to display on the mini-followup page.
